### PR TITLE
enable nestif linter

### PR DESCRIPTION
**Pull request by elevran**
- uncomment linter
- simplify nested if/else to align happy path to left where possible

Note to reviewer: **please double check that I did not change the logic!**

### CODE REVIEW

**Review by kfirtoledo — APPROVED** [review]

File: pkg/platform/k8s/reconciler.go [hunk 1]
```diff
@@ -54,31 +54,30 @@ func (r *Reconciler) CreateResource(name string, obj client.Object) {
...
+	if !ok {
+		r.logger.Errorf("error occurred while updating K8s %v: %v is not exist", reflect.TypeOf(obj).String(), name)
+	}
+
+	if obj.GetName() == oldObj.GetName() {
+		err := r.client.Update(context.Background(), obj)
+		if err != nil {
+			r.logger.Errorf("error occurred while updating K8s %v %v:", reflect.TypeOf(obj).String(), err)
+			r.failedList[name] = reconcileObj{spec: obj, op: updateOp}
 			return
```
Comment (orozery) [inline pkg/platform/k8s/reconciler.go:67]: Why did you move the return inside the error case?

Comment (elevran) [reply pkg/platform/k8s/reconciler.go:67]: should processing (create resource) continue when the Update failed? Does it depend on the error returned?
Perhaps I misunderstood the logic... Let me try to revert this portion by removing the return and see if tests pass
